terraform/terraform-modules/cognito-auth/cognito-auth.tf

```diff
@@ -37,58 +37,92 @@ resource "aws_lb_target_group" "mwaa_auth_alb_lambda_tg" {
   target_type                        = "lambda"
 }
 
-
 data "aws_caller_identity" "current" {}
-
 data "aws_region" "current" {}
 
+# Automatically build Lambda package before zipping
 resource "null_resource" "install_dependencies" {
   provisioner "local-exec" {
-    command = "bash ${path.module}/lambda/build-lambda.sh"
-    # Ensure Docker is installed and running
-    environment = {
-      path_module = path.module
-    }
+    command = <<EOT
+      set -e
+      echo "Building Lambda package automatically..."
+      cd ${path.module}/lambda
```

Comment:
In case `path.module` contains spaces, consider quoting the `cd`:

    cd "${path.module}/lambda"